lib/bolt/util.rb

```diff
@@ -179,6 +179,29 @@ def deep_clone(obj, cloned = {})
         end
       end
 
+      # This is stubbed for testing validate_file
+      def file_stat(path)
+        File.stat(path)
+      end
+
+      def validate_file(type, path, allow_dir = false)
+        stat = file_stat(path)
```

Comment:
Can we safely `file_stat(File.expand_path(path))` here? When I use a `~` for my homedir `File.stat` immediately throws `ENOENT` and warning is raised in `Bolt::Transport::SSH::Connection`. 
```
cas@cas-ThinkPad-T460p:~/working_dir/bolt$ bolt command run whoami -t pooler
Started on s5xx09est930z86.delivery.puppetlabs.net...
The ssh key '~/.ssh/id_rsa-acceptance' does not exist
Finished on s5xx09est930z86.delivery.puppetlabs.net:
  STDOUT:
    root
Successful on 1 node: pooler
Ran on 1 node in 0.35 seconds
```